js/app/modules/layout/responsiveMargins.js

```diff
@@ -124,6 +124,8 @@ const ResponsiveMargins = new Module.Class({
         let margins = this._get_responsive_margins();
         let [min_size, nat_size] = this.get_preferred_size();
         let base_min_width = min_size.width - margins.tiny.left - margins.tiny.right;
+        if (this.get_request_mode() === Gtk.SizeRequestMode.HEIGHT_FOR_WIDTH)
```

Comment:
This whole bug is nasty.

Not totally sure I'm understanding what this is trying to do. Could you walk me through the change and what you thinks happening with our allocation cycles? I can see that in height-for-width sizing the nat height will often be smaller than min height here, but it seems incorrect to just assume that here.

A lot of this widget seems to be playing fast and lose height for width sizing, and seems like we might want to just rewrite a bit of it. And long term, I'd really love to see this widget die and be replaced by much better responsive tools.

Another fun fact is that in the search page case you contain a `card/list` a widget which stomps on its own allocation without any attempt to update its preferred width of heights requests. It definitely breaks all sorts of assumptions in gtk and could be contributing to the problems here.